### PR TITLE
Nav list component automated tests

### PR DESCRIPTION
**Pull request by martypowell**
Tests to see if nav-list component is working properly.

A good page for this is, https://www.baltimorecountymd.gov/Agencies/economicdev/business/index.html

image

### CODE REVIEW

**Comment by martypowell** [comment]
As referenced on #7

File: spec/components/nav-list/nav-list.automater.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
+const dateParse = require('date-fns/parse');
+const isDateEqual = require('date-fns/is_equal');
+const startOfDay = require('date-fns/start_of_day');
+const webdriver = require('selenium-webdriver');
+
+const seleniumProxy = require('../../selenium-proxy');
+
+const By = webdriver.By;
+
+const statuses = {
```
Comment (mxsnyder) [inline spec/components/nav-list/nav-list.automater.js]: Should these be moved to the constants file?

File: spec/components/nav-list/nav-list.automater.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		}
+		catch (ex) {
+			return false;
+		}
+	};
+
+	const checkNavPanelStatus = async (initialStatus, targetButtonToClickSelector) => {
+		const collapsedButtons = await state.driver.findElements(By.css(targetButtonToClickSelector));
+		if (collapsedButtons && collapsedButtons.length) {
+			const collapsedButton = collapsedButtons[0];
```
Comment (mxsnyder) [inline spec/components/nav-list/nav-list.automater.js]: Why are using index 0? I think I know, but you should either add a comment or make the index a const named something like "firstCollapsedButtonIndex".

File: spec/components/nav-list/nav-list.automater.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+	const doesNavCollapse = async () => {
+		return await checkNavPanelStatus(statuses.EXPANDED, state.options.collapseButtonSelector);
+	};
+
+	const doesNavExpand = async () => {
+		return await checkNavPanelStatus(statuses.COLLAPSED, state.options.expandButtonSelector);
+	};
+
+
```
Comment (mxsnyder) [inline spec/components/nav-list/nav-list.automater.js]: You can reduce your whitespace here.

File: spec/components/nav-list/nav-list.automater.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	const doesNavExpand = async () => {
+		return await checkNavPanelStatus(statuses.COLLAPSED, state.options.expandButtonSelector);
+	};
+
+	
+
+	const doesNavShowRightCaretWhenCollapsed = async () => {
+		const expandButton = await state.driver.findElements(By.css(state.options.expandButtonSelector));
+		if (expandButton && expandButton.length) {
+			const expandButtonClasses = await expandButton[0].getAttribute('class');
```
Comment (mxsnyder) [inline spec/components/nav-list/nav-list.automater.js]: Same comment about index 0. Just assume I made it elsewhere too.

**Review by mxsnyder — CHANGES_REQUESTED** [review]
Minor suggestions, mostly about changing indexes to variables.

File: spec/components/nav-list/nav-list.automater.js (hunk 1 above)
Comment (martypowell) [reply spec/components/nav-list/nav-list.automater.js]: good call, done

File: spec/components/nav-list/nav-list.automater.js (hunk 2 above)
Comment (martypowell) [reply spec/components/nav-list/nav-list.automater.js]: good call, done.

File: spec/components/nav-list/nav-list.automater.js (hunk 4 above)
Comment (martypowell) [reply spec/components/nav-list/nav-list.automater.js]: done

File: spec/components/nav-list/nav-list.automater.js (hunk 3 above)
Comment (martypowell) [reply spec/components/nav-list/nav-list.automater.js]: done

**Comment by martypowell** [comment]
@mxsnyder ready for another review

**Review by mxsnyder — APPROVED** [review]

**Pull request merged**